docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
```

Comment:
This feels like something that would be better linked off to. Maybe we can add a link to...

- https://developers.google.com/identity/protocols/OAuth2ServiceAccount or
- https://cloud.google.com/storage/docs/authentication#service_accounts ?

I like the second one better, but it's under the bucket of "Cloud Storage" so could possibly confuse people... (But it has screen shots... which is nice...)

I'm going to add the second one -- feel free to change if necessary.